src/main/java/io/strimzi/kafka/metrics/http/HttpServers.java

```diff
@@ -53,22 +53,33 @@ public synchronized static void release(ServerCounter serverCounter) {
      * Class used to keep track of the HTTP server started on a listener.
      */
     public static class ServerCounter {
+        private HTTPServer.Builder builder;
+        private Listener listener;
+        private AtomicInteger count;
+        private HTTPServer server;
 
-        private final AtomicInteger count;
-        private final HTTPServer server;
-        private final Listener listener;
-
-        private ServerCounter(Listener listener, PrometheusRegistry registry) throws IOException {
-            this.count = new AtomicInteger();
-            HTTPServer.Builder builder = HTTPServer.builder()
-                    .port(listener.port)
-                    .registry(registry);
+        private ServerCounter(Listener listener, PrometheusRegistry registry) {
+            this.builder = HTTPServer.builder()
+                .port(listener.port)
+                .registry(registry);
             if (!listener.host.isEmpty()) {
                 builder.hostname(listener.host);
             }
-            this.server = builder.buildAndStart();
-            LOG.debug("Started HTTP server on http://{}:{}", listener.host, server.getPort());
             this.listener = listener;
+            this.count = new AtomicInteger();
+        }
+
+        /**
+         * Start the HTTP server.
+         */
+        public void start() {
```

Comment:
In `PrometheusMetricsReporterConfig` you used the `@SuppressFBWarnings("CT_CONSTRUCTOR_THROW")` annotation to silence this Spotbug warning, but here you decided to wrap `IOException` into an unchecked exception instead.
Can you explain why the 2 different approaches?
Also if you decide to keep the `start()` method, it should be private.

Reply:
You can't throw an exception from the constructor, that what needs to change to avoid the finalizer attack.

> Can you explain why the 2 different approaches?

To me it seems natural to create and then start a server (two separate operations), while the allow list compilation seems more related to initialization. For `PrometheusMetricsReporterConfig`, a good alternative could be to lazy compile the allow list on the very first call to `isAllowed`. Wdyt?

> Also if you decide to keep the start() method, it should be private.

Done.

Reply:
> For PrometheusMetricsReporterConfig, a good alternative could be to lazy compile the allow list on the very first call to isAllowed. Wdyt?

I prefer that the reporter fails to startup if a bad allowlist is provided instead of failing at some random point in the future when the `/metrics` endpoint is scrapped.

Reply:
> To me it seems natural to create and then start a server (two separate operations), while the allow list compilation seems more related to initialization. 

Fair enough, but why are we wrapping `IOException` into an unchecked exception if it's not thrown by the constructor anymore?